modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc

```diff
@@ -1,18 +1,62 @@
-(ns {{main-ns}}.core
+(ns {{main-ns}} .core
   (:gen-class)
   (:require
-   [fierycod.holy-lambda.response :as hr]
-   [fierycod.holy-lambda.native-runtime :as native]
-   [fierycod.holy-lambda.agent :as agent]
-   [fierycod.holy-lambda.core :as h]))
+    [fierycod.holy-lambda.response :as hr]
+    [fierycod.holy-lambda.interceptor :as i]
+    [fierycod.holy-lambda.native-runtime :as native]
+    [fierycod.holy-lambda.agent :as agent]
+    [fierycod.holy-lambda.core :as h]))
 
-(h/deflambda ExampleLambda <
-  [{:keys [event ctx]}]
-  #?(:bb
-     (hr/text (str "Hello world. Babashka is sweet friend of mine! Babashka version: " (System/getProperty "babashka.version")))
-     :clj (hr/text "Hello world")))
 
+;; optionally, provide runtime specific implementations if needed
+(defn say-hello
+  []
+  #?(:bb  (str "Hello world. Babashka is a sweet friend of mine! Babashka version: " (System/getProperty "babashka.version"))
+     :clj "Hello world"))
+
+
+;; An interceptor is a pair of unary functions representing inbound requests and outbound responses.
+;;
+;; holy-lambda calls the :enter function on the way "in" to handling a request. It calls the :leave function on the way back "out".
+;;
+;; The :enter function is called with the request map and must return either a request map or a channel that will deliver a request map.
+;; Similarly, the :leave function is called with the response map and must return either a response map or a channel that will deliver a response map.
+;;
+;; Interceptors may be chained together:
+;;     < {:interceptors [Interceptor1 Interceptor2]}
+;;
```

Comment:
There is also another way of joining interceptors together.
```
(def mixin1 {:interceptors [interceptor1]})
(def mixin2 {:interceptors [interceptor2]})

(def mixin3 (h/merge-mixins mixin1 mixin2)) ;; => {:interceptors [interceptor1 interceptor2]}
```

Reply:
added to docs